src/main/daml/Daml/Finance/Instrument/Bond/Util.daml

```diff
@@ -59,11 +61,18 @@ createCallableBondClaims dateToTime schedule periodicSchedule useAdjustedDatesFo
     -- claim (in front of the or node, see explanation below).
     principal = give notionalAmount
 
-    -- Currently, it is possible to call on every coupon date
-    callPossibleInit = replicate (length schedule) True
+    -- Find out on which coupon dates it is possible to call the bond.
+    callPossibleInit = map (\s -> isSome $
+      find (\c -> c.unadjustedEndDate == s.unadjustedEndDate) callableSchedule)
```

Comment:
Is it possible for the two schedules to have different calendars, so that unadjusted date is the same but adjusted could be different?

Reply:
I realize that you might also be asking whether the user could specify two different schedules and thereby ending up with an incorrect claims tree. No, that is not possible, because the same `holidayCalendarIds` is used for both schedules. Also, to ensure that no potential call dates are accidentally ignored (by a mis-specified call schedule), there is a check for that as well.

Reply:
However, I wonder whether it would be useful to do this check on adjusted dates anyway, just to future proof it. There is probably no downside to it that I can think of at the moment.

Reply:
I have changed it to adjusted dates now.